### PR TITLE
SONARPY-142 Migrate plugin on SQ 5.6 LTS

### PR DESCRIPTION
**Pull request by ivandalbosco**

### CODE REVIEW

**Comment by vilchik-elena** [comment]
@ivandalbosco could you update PR title?

File: its/ruling/src/test/java/org/sonar/python/it/PythonRulingTest.java [hunk 1]
```diff
@@ -44,9 +46,11 @@
 
   @Test
   public void test() throws Exception {
+    ImmutableList<String> pipo = ImmutableList.of();
+    System.out.println(pipo.isEmpty());
```
Comment (vilchik-elena) [inline its/ruling/src/test/java/org/sonar/python/it/PythonRulingTest.java]: @ivandalbosco System.out.println

File: pom.xml [hunk 2]
```diff
@@ -5,7 +5,7 @@
   <parent>
     <groupId>org.sonarsource.parent</groupId>
     <artifactId>parent</artifactId>
-    <version>31</version>
+    <version>36</version>
```
Comment (vilchik-elena) [inline pom.xml]: @ivandalbosco For future: better update parent in separate commit

File: pom.xml [hunk 3]
```diff
@@ -155,4 +203,39 @@
     </profile>
   </profiles>
 
+  <build>
```
Comment (vilchik-elena) [inline pom.xml]: @ivandalbosco What is it for?

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSquidSensorTest.java [hunk 4]
```diff
@@ -73,15 +73,16 @@ public void should_execute_on_python_project() {
     Project project = mock(Project.class);
     assertThat(sensor.toString()).isEqualTo("PythonSquidSensor");
     assertThat(sensor.shouldExecuteOnProject(project)).isFalse();
-    fs.add(new DefaultInputFile("test.py").setLanguage(Python.KEY));
+    fs.add(new DefaultInputFile(TestUtils.MODULE, "test.py").setLanguage(Python.KEY));
```
Comment (vilchik-elena) [inline sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSquidSensorTest.java]: @ivandalbosco imho this module key parameter is not important in unit tests. WDYT about dropping TestUtils (here and in squid module) and passing e.g. `"moduleKey"` instead `TestUtils.MODULE`?
p.s. TestUtils contains method which is not used

File: its/pom.xml [hunk 5]
```diff
@@ -29,9 +29,7 @@
       <build>
         <plugins>
           <plugin>
-            <groupId>org.apache.maven.plugins</groupId>
             <artifactId>maven-dependency-plugin</artifactId>
-            <version>2.10</version>
```
Comment (vilchik-elena) [inline its/pom.xml]: @ivandalbosco why did you drop groupId and version?

File: its/plugin/src/test/java/com/sonar/python/it/plugin/TestReportTest.java [hunk 6]
```diff
@@ -26,6 +26,8 @@
 import com.sonar.orchestrator.Orchestrator;
 import com.sonar.orchestrator.build.BuildResult;
 import com.sonar.orchestrator.build.SonarRunner;
+
+import org.junit.Before;
```
Comment (vilchik-elena) [inline its/plugin/src/test/java/com/sonar/python/it/plugin/TestReportTest.java]: @ivandalbosco which settings for import formatting do you have? It should be like this:
import all other imports
<blank line>
import static all other imports

(I'm asking because i see blank line on previous line)

File: its/ruling/src/test/java/org/sonar/python/it/PythonRulingTest.java (hunk 1 above)
Comment (ivandalbosco) [inline its/ruling/src/test/java/org/sonar/python/it/PythonRulingTest.java]: Hmmm... sorry

File: pom.xml (hunk 2 above)
Comment (ivandalbosco) [inline pom.xml]: OK. Will try to remember.

File: pom.xml (hunk 3 above)
Comment (ivandalbosco) [inline pom.xml]: I'll have to remove it later on. It is a problem with Eclipse. Eclipse fails loading the project when it does not know what to do with a plugin - here with the licence-mave-plugin. It attempted to install Sonar's m2e-error-disabler-plugin, but Eclipse failed (NPE...) doing so. Will try with new yearly Eclipse version, expected to come out today.

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSquidSensorTest.java (hunk 4 above)
Comment (ivandalbosco) [inline sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSquidSensorTest.java]: I did not know what to do with moduleKey, so I centralized its value in one place.

File: its/pom.xml (hunk 5 above)
Comment (ivandalbosco) [inline its/pom.xml]: @vilchik-elena 
groupId is not required for Maven native plugins.
the version of all Maven plugins is inherited from the parent POM org.sonarsource.parent:parent

File: its/plugin/src/test/java/com/sonar/python/it/plugin/TestReportTest.java (hunk 6 above)
Comment (ivandalbosco) [inline its/plugin/src/test/java/com/sonar/python/it/plugin/TestReportTest.java]: @vilchik-elena You are right, we need to have identical settings. I am using the preferences from https://github.com/SonarSource/sonar-developer-toolset. Looks like the basic Eclipse settings are not modified by the SonarSource prefs. We should check that together, because it causes noisy changes.

File: python-squid/src/test/java/org/sonar/python/FileLinesVisitorTest.java [hunk 7]
```diff
@@ -44,11 +45,11 @@
   @Test
   public void test() {
     FileLinesContextFactory fileLinesContextFactory = mock(FileLinesContextFactory.class);
-    DefaultFileSystem fileSystem = new DefaultFileSystem();
+    DefaultFileSystem fileSystem = new DefaultFileSystem(Paths.get(""));
     FileLinesContext fileLinesContext = mock(FileLinesContext.class);
 
     File file = new File(BASE_DIR, "file_lines.py");
-    InputFile inputFile = new DefaultInputFile(file.getPath());
+    DefaultInputFile inputFile = new DefaultInputFile(TestUtils.MODULE, file.getPath());
```
Comment (vilchik-elena) [inline python-squid/src/test/java/org/sonar/python/FileLinesVisitorTest.java]: @ivandalbosco We can replace `TestUtils.MODULE` with `""` or `"moduleKey"`

File: sonar-python-plugin/pom.xml [hunk 8]
```diff
@@ -49,35 +43,54 @@
...
+              <goal>shade</goal>
+            </goals>
+            <configuration>
+              <artifactSet>
+                <excludes>
+                  <exclude>cglib:cglib-nodep</exclude>
+                </excludes>
+              </artifactSet>
+              <createDependencyReducedPom>false</createDependencyReducedPom>
+              <createSourcesJar>true</createSourcesJar>
```
Comment (vilchik-elena) [inline sonar-python-plugin/pom.xml]: @ivandalbosco i don't see effect of this line. Do you?

File: its/plugin/src/test/java/com/sonar/python/it/plugin/TestReportTest.java (hunk 6 above)
Comment (ivandalbosco) [inline its/plugin/src/test/java/com/sonar/python/it/plugin/TestReportTest.java]: @vilchik-elena OK, changed with new imported Eclipse profile

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSquidSensorTest.java [hunk 9]
```diff
@@ -73,15 +73,16 @@ public void should_execute_on_python_project() {
...
   }
 
   @Test
   public void should_analyse() {
     String relativePath = "src/test/resources/org/sonar/plugins/python/code_chunks_2.py";
-    DefaultInputFile inputFile = new DefaultInputFile(relativePath).setLanguage(Python.KEY);
-    inputFile.setAbsolutePath((new File(relativePath)).getAbsolutePath());
+    DefaultInputFile inputFile = new DefaultInputFile(TestUtils.MODULE, relativePath).setLanguage(Python.KEY);
+    
+    inputFile.setModuleBaseDir(Paths.get("").toAbsolutePath());
```
Comment (vilchik-elena) [inline sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSquidSensorTest.java]: @ivandalbosco We can refactor `Paths.get("")` into this class' private static field

Comment (ivandalbosco) [inline sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSquidSensorTest.java]: @vilchik-elena If you want...

File: its/ruling/src/test/java/org/sonar/python/it/PythonRulingTest.java [hunk 10]
```diff
@@ -29,11 +29,13 @@
...
 
 import com.google.common.collect.ImmutableList;
-import com.google.common.io.Files;
+
+
 import com.sonar.orchestrator.Orchestrator;
 import com.sonar.orchestrator.build.SonarRunner;
 import com.sonar.orchestrator.locator.FileLocation;
 
+import com.google.common.io.Files;
```
Comment (vilchik-elena) [inline its/ruling/src/test/java/org/sonar/python/it/PythonRulingTest.java]: @ivandalbosco again there is weird imports formatting here

Comment (ivandalbosco) [inline its/ruling/src/test/java/org/sonar/python/it/PythonRulingTest.java]: @vilchik-elena I suggest I create a ticket to format all classes in one commit. Should solve the issue permanently

File: pom.xml (hunk 3 above)
Comment (vilchik-elena) [inline pom.xml]: @ivandalbosco ok. Could you remove it now, otherwise i can't merge PR.

File: its/ruling/src/test/java/org/sonar/python/it/PythonRulingTest.java (hunk 10 above)
Comment (vilchik-elena) [inline its/ruling/src/test/java/org/sonar/python/it/PythonRulingTest.java]: @ivandalbosco no need for ticket. You can do that by creating a separate PR just after we will merge this one.

File: sonar-python-plugin/pom.xml (hunk 8 above)
Comment (vilchik-elena) [inline sonar-python-plugin/pom.xml]: @ivandalbosco ping

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/TestUtils.java [hunk 11]
```diff
@@ -35,5 +35,5 @@ public static File loadResource(String resourceName) {
 
     return resourceAsFile;
   }
-
+
```
Comment (vilchik-elena) [inline sonar-python-plugin/src/test/java/org/sonar/plugins/python/TestUtils.java]: @ivandalbosco Please try to avoid such changeless changes in future

File: sonar-python-plugin/pom.xml [hunk 12]
```diff
@@ -49,35 +43,54 @@
...
+        <executions>
+          <execution>
+            <phase>package</phase>
+            <goals>
+              <goal>shade</goal>
+            </goals>
+            <configuration>
+              <artifactSet>
+                <excludes>
+                  <exclude>cglib:cglib-nodep</exclude>
```
Comment (vilchik-elena) [inline sonar-python-plugin/pom.xml]: @ivandalbosco Why this block?
